src-test/src/com/etendoerp/dependencymanager/process/DependencyTreeBuilderTest.java

```diff
@@ -0,0 +1,106 @@
+
+    Package package1 = DependencyManagerTestUtils.createPackage(PACKGE_1, GROUP_COM_ETENDOERP);
+    Package package2 = DependencyManagerTestUtils.createPackage(PACKGE_2, GROUP_COM_ETENDOERP);
+    Package package3 = DependencyManagerTestUtils.createPackage(PACKGE_3, GROUP_COM_ETENDOERP);
+
+    PackageVersion packageVersion1 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package1, FROM_CORE,
+        LATEST_CORE);
+    PackageVersion packageVersion2 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package2, FROM_CORE,
+        LATEST_CORE);
+    PackageVersion packageVersion3 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package3, FROM_CORE,
+        LATEST_CORE);
+
+    DependencyManagerTestUtils.createPackageDependency(packageVersion1, packageVersion2, "module.jar1",
+        GROUP_COM_ETENDOERP, FIRST_VERSION, false);
+    DependencyManagerTestUtils.createPackageDependency(packageVersion2, packageVersion3, "module.jar2",
+        GROUP_COM_ETENDOERP, FIRST_VERSION, false);
+
+    OBDal.getInstance().flush();
+    OBDal.getInstance().refresh(packageVersion1);
+    OBDal.getInstance().refresh(packageVersion2);
+
+    List<PackageDependency> dependencyList = DependencyTreeBuilder.createDependencyTree(packageVersion1);
+    assertEquals(1, packageVersion1.getETDEPPackageDependencyList().size());
+    assertEquals(2, dependencyList.size());
+  }
+
+  @Test
+  public void testRemoveEtendoCoreDependencyFromPackageVersion() {
+    Package package1 = DependencyManagerTestUtils.createPackage(PACKGE_1, GROUP_COM_ETENDOERP);
+    Package package2 = DependencyManagerTestUtils.createPackage(PACKGE_2, GROUP_COM_ETENDOERP);
+    Package package3 = DependencyManagerTestUtils.createPackage(PACKGE_3, GROUP_COM_ETENDOERP);
+
+    PackageVersion packageVersion1 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package1, FROM_CORE,
+        LATEST_CORE);
+    PackageVersion packageVersion2 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package2, FROM_CORE,
+        LATEST_CORE);
+    PackageVersion packageVersion3 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package3, FROM_CORE,
+        LATEST_CORE);
+
+    DependencyManagerTestUtils.createPackageDependency(packageVersion1, packageVersion2, "module.jar1",
+        GROUP_COM_ETENDOERP, FIRST_VERSION, false);
+    DependencyManagerTestUtils.createPackageDependency(packageVersion1, packageVersion3, "etendo-core",
+        GROUP_COM_ETENDOERP, FIRST_VERSION, false);
+
+    OBDal.getInstance().flush();
+    OBDal.getInstance().refresh(packageVersion1);
+    OBDal.getInstance().refresh(packageVersion2);
+
+    int initialDependencyCount = packageVersion1.getETDEPPackageDependencyList().size();
+    List<PackageDependency> dependencyList = DependencyTreeBuilder.createDependencyTree(packageVersion1);
+    assertEquals(2, initialDependencyCount);
+    assertEquals(1, dependencyList.size());
+  }
```

Comment:
The code for the tests testFindDependenciesFromPackageVersion and testRemoveEtendoCoreDependencyFromPackageVersion seems to be very similar, let's try to modularize it and avoid duplication, I can think of something like:

```
public class DependencyManagerTest {

  private PackageVersion setupPackageVersions() {
    Package package1 = DependencyManagerTestUtils.createPackage(PACKGE_1, GROUP_COM_ETENDOERP);
    Package package2 = DependencyManagerTestUtils.createPackage(PACKGE_2, GROUP_COM_ETENDOERP);
    Package package3 = DependencyManagerTestUtils.createPackage(PACKGE_3, GROUP_COM_ETENDOERP);

    PackageVersion packageVersion1 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package1, FROM_CORE, LATEST_CORE);
    PackageVersion packageVersion2 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package2, FROM_CORE, LATEST_CORE);
    PackageVersion packageVersion3 = DependencyManagerTestUtils.createPackageVersion(FIRST_VERSION, package3, FROM_CORE, LATEST_CORE);

    OBDal.getInstance().flush();
    OBDal.getInstance().refresh(packageVersion1);
    OBDal.getInstance().refresh(packageVersion2);

    return packageVersion1;
  }

  private void createDependencies(PackageVersion packageVersion1, String module1, PackageVersion packageVersion2, String module2, PackageVersion packageVersion3) {
    DependencyManagerTestUtils.createPackageDependency(packageVersion1, packageVersion2, module1, GROUP_COM_ETENDOERP, FIRST_VERSION, false);
    DependencyManagerTestUtils.createPackageDependency(packageVersion2, packageVersion3, module2, GROUP_COM_ETENDOERP, FIRST_VERSION, false);
  }

  @Test
  public void testFindDependenciesFromPackageVersion() {
    PackageVersion packageVersion1 = setupPackageVersions();

    createDependencies(packageVersion1, "module.jar1", packageVersion1.getETDEPPackageDependencyList().get(0).getPackageVersionDependency(), "module.jar2", packageVersion1.getETDEPPackageDependencyList().get(1).getPackageVersionDependency());

    List<PackageDependency> dependencyList = DependencyTreeBuilder.createDependencyTree(packageVersion1);
    assertEquals(1, packageVersion1.getETDEPPackageDependencyList().size());
    assertEquals(2, dependencyList.size());
  }

  @Test
  public void testRemoveEtendoCoreDependencyFromPackageVersion() {
    PackageVersion packageVersion1 = setupPackageVersions();

    createDependencies(packageVersion1, "module.jar1", packageVersion1.getETDEPPackageDependencyList().get(0).getPackageVersionDependency(), "etendo-core", packageVersion1.getETDEPPackageDependencyList().get(1).getPackageVersionDependency());

    int initialDependencyCount = packageVersion1.getETDEPPackageDependencyList().size();
    List<PackageDependency> dependencyList = DependencyTreeBuilder.createDependencyTree(packageVersion1);
    assertEquals(2, initialDependencyCount);
    assertEquals(1, dependencyList.size());
  }
}
```